arch/arm/core/irq_manage.c

```diff
@@ -132,6 +132,25 @@ void z_irq_spurious(void *unused)
 	__reserved();
 }
 
+static int z_arm_irq_current_prio(void)
+{
+	u32_t isr_vector_num = __get_IPSR() & IPSR_ISR_Msk;
+	s32_t irq_type;
+
+	if (isr_vector_num == 0) {
+		return INT_MAX;
+	}
+
+	irq_type = ((s32_t)isr_vector_num - 16);
+
+	return (NVIC_GetPriority((IRQn_Type)irq_type) & 0xFF);
+}
+
+bool z_arm_irq_can_preempt(u32_t irq_line)
+{
+	return z_arm_irq_current_prio() > NVIC_GetPriority((IRQn_Type)irq_line);
```

Comment:
Is this really what we want, here? What if the irq_line is, in fact, not enabled, and the priority is a trash value? Can this actually happen?

Reply:
current use case is using it in the driver which handles `irq_line`. I wouldn't involve enabling checking into that. What to return if interrupt is disabled? `false` would be correct because disabled interrupt cannot be preempt current context but that is not what we wanted to check.

By default priority in nvic is set to 0 (highest). I can add a note that only priority is checked and not the state of the interrupt.

Reply:
This all depends on what API we want to provide, see the discussion below. So, if the irq_line is not enabled, I think this function should return false.